pkg/api/server/v1alpha2/auth/rbac.go

```diff
@@ -166,11 +167,11 @@ func (r *RBAC) Check(ctx context.Context, namespace, resource, verb string) erro
 	return status.Error(codes.Unauthenticated, retMsg)
 }
 
-// convertExtra converts the map[string][]string to map[string]ExtraValue for Subject Access Review.
-func convertExtra(extra map[string][]string) map[string]authzv1.ExtraValue {
+// convertExtra converts the map[string][]string or authnv1.ExtraValue to map[string]ExtraValue for Subject Access Review.
+func convertExtra[T []string | authnv1.ExtraValue](extra map[string]T) map[string]authzv1.ExtraValue {
```

Comment:
ooo, go generics! Good godoc here for folks who are unfamiliar with how go generics work.